code/modules/mob/living/living_dreams.dm

```diff
@@ -0,0 +1,14 @@
+/mob/living
+	var/dreaming = FALSE
+
+/mob/living/proc/handle_dreams()
+	set waitfor = FALSE
+	if(!client || dreaming || !prob(5))
+		return
+	dreaming = TRUE
+	for(var/i = 1 to rand(1,4))
+		to_chat(src, SPAN_NOTICE("<i>... [pick(SSlore.dreams)] ...</i>"))
```

Comment:
Potential future change: make the random-dream-message-getting a proc on mob/living so that things like stressors can cause nightmares, make you dream about specific topics, etc.